### PR TITLE
VDS-226: Add customer segment preview

### PR DESCRIPTION
**Pull request by asvishnyakov**

### CODE REVIEW

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/resources/customerSegments.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+angular.module('virtoCommerce.DemoSolutionFeaturesModule')
+.factory('virtoCommerce.DemoSolutionFeaturesModule.customerSegmentsApi', ['$resource', function ($resource) {
+    return $resource('api/demo/customersegments/:id', {}, {
+        new: { method: 'GET', url: 'api/demo/customersegments/new' },
+        update: { method: 'PUT' },
+        delete: { method: 'POST', url: 'api/demo/customersegments/delete'},
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/resources/customerSegments.js]: typo? method should be: 'DELETE'

Comment (asvishnyakov) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/resources/customerSegments.js]: Fixed

**Review by pushnitsa — DISMISSED** [review]

**Review by trueboroda — DISMISSED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
     var blade = $scope.blade;
     blade.toolbarCommands = [
         {
-            name: "platform.commands.preview", icon: 'fa fa-eye'
+            name: "platform.commands.preview", icon: 'fa fa-eye',
+            executeMethod: (currentBlade) => {
+                let properties = currentBlade.currentEntities
+                    .filter(property => property.valueType === 'Boolean' || property.values.length && property.values.every(value => !!value.value))
+                    .reduce((propertiesAggregate, property) => {
+                        propertiesAggregate[property.name] = property.values && property.values.length ? property.values.map(propertyValue => propertyValue.value.name || propertyValue.value) : [false];
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js:12]: Please add a short comment why we doing so.

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js:12]: Done

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**